### ISSUE TITLE
gnt-backup export does not work with "lvm based" extstorage

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1168]
Originally reported of Google Code with ID 1112.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

gnt-cluster (ganeti v2.11.6) 2.11.6
Software version: 2.11.6
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.6
root@xen06:~# hspace --version
hspace (ganeti) version v2.11.6
compiled with ghc 7.6
running on linux x86_64

<b>What distribution are you using?</b>
Debian Jessie 8.1

<b>What steps will reproduce the problem?</b>
1.gnt-backup export -n NODE INSTANCE
<b>2.</b>
<b>3.</b>

<b>What is the expected output? What do you see instead?</b>
Something like... "Exporting ... done."

Ganeti output:

Fri Jun 26 11:44:01 2015 Shutting down instance debian.rkt-zentral.local
Fri Jun 26 11:44:01 2015 Creating a snapshot of disk/0 on node xen06.rkt-zentral.local
Fri Jun 26 11:44:02 2015  - WARNING: Could not snapshot disk/0 on node xen06.rkt-zentral.local: Cannot snapshot non-lvm block device '('lvm', 'dfd2c6da-90ba-4c9c-96d2-fe87e215cf53.ext.disk0')' of type 'ext'
Fri Jun 26 11:44:02 2015 Finalizing export on xen06.rkt-zentral.local
Fri Jun 26 11:44:02 2015  - WARNING: Could not finalize export for instance debian.rkt-zentral.local on node xen06.rkt-zentral.local: Error while executing backend function: [Errno 2] No such file or directory: '/var/lib/ganeti/export/debian.rkt-zentral.local.new/config.iniR_eBfn.new'
Failure: command execution error:
Export failed, errors in export finalization, disk export: disk(s) 0

<b>Please provide any additional information below.</b>
Ganeti documentation for 2.11.x says that gnt-backup export for extstorage is supported when "snapshot" provider script is available.
Some while ago I have written my own "lvm based" extstorage provider which you can find at the bottom here... https://code.google.com/p/ganeti/wiki/ExtStorageProviders
The "snapshot" script is included there and "should" work when used.
Right now I use my own backup scripts but wanted to give gnt-backup a try.
My "snapshot" script should write some information to var.log inside provider directory, but nothing appears there.
So it seems that "gnt-backup export" did not use it and the documentation for ganeti 2.11.x is wrong or "gnt-backup export" broken.

By the way... gnt-backup import to extstorage WORKS!
(The export was taken when the instance was created with plain disk-template)

gnt-backup import -t ext --disk 0:size=8G,provider=lvm,vgname=storage8 -n xen06 --no-ip-check --no-name-check debian.rkt-zentral.local

Fri Jun 26 12:30:42 2015 * disk 0, size 8.0G
Fri Jun 26 12:30:42 2015 * creating instance disks...
Fri Jun 26 12:30:43 2015 adding instance debian.rkt-zentral.local to cluster config
Fri Jun 26 12:30:45 2015  - INFO: Waiting for instance debian.rkt-zentral.local to sync disks
Fri Jun 26 12:30:45 2015  - INFO: Instance debian.rkt-zentral.local's disks are in sync
Fri Jun 26 12:30:45 2015 * running the instance OS import scripts...
Fri Jun 26 12:30:45 2015 Exporting disk/0 from xen06.rkt-zentral.local to xen06.rkt-zentral.local
Fri Jun 26 12:30:49 2015 disk/0 is now listening, starting export
Fri Jun 26 12:30:53 2015 disk/0 is receiving data on xen06.rkt-zentral.local
Fri Jun 26 12:30:53 2015 disk/0 is sending data on xen06.rkt-zentral.local
Fri Jun 26 12:31:03 2015 disk/0 sent 170M, 15.4 MiB/s, 6%, ETA 2m 35s
Fri Jun 26 12:32:06 2015 disk/0 sent 1.2G, 18.7 MiB/s, 49%, ETA 1m 9s
Fri Jun 26 12:33:08 2015 disk/0 sent 1.7G, 8.2 MiB/s, 69%, ETA 1m 37s
Fri Jun 26 12:34:11 2015 disk/0 sent 2.4G, 10.8 MiB/s, 96%, ETA 8s
Fri Jun 26 12:34:42 2015 disk/0 finished receiving data
Fri Jun 26 12:34:43 2015 disk/0 finished sending data
Fri Jun 26 12:34:43 2015 Running rename script for debian.rkt-zentral.local

Kind regards
Darius

```
Originally added on 2015-06-26 10:53:28 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
In the meantime I've found out, that the Snapshot definition for extstorage was implemented in ganeti 2.14.0.

Thanks.

This Issue can be closed.

Kind regards
Darius


```
Originally added on 2015-06-26 16:03:07 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-06-26 16:12:01 +0000 UTC.

Changed State: Invalid
